examples/tool_call/notion_toolkit.py

```diff
@@ -0,0 +1,113 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+from camel.agents import ChatAgent
+from camel.configs.openai_config import ChatGPTConfig
+from camel.messages import BaseMessage
+from camel.models import ModelFactory
+from camel.toolkits import NotionToolkit
+from camel.types import ModelPlatformType, ModelType
+
+# Define system message
+sys_msg = BaseMessage.make_assistant_message(
+    role_name="Tools calling opertor", content="You are a helpful assistant"
+)
+
+# Set model config
+tools = NotionToolkit().get_tools()
+model_config_dict = ChatGPTConfig(
+    temperature=0.0,
+).as_dict()
+
+model = ModelFactory.create(
+    model_platform=ModelPlatformType.OPENAI,
+    model_type=ModelType.GPT_4O_MINI,
+    model_config_dict=model_config_dict,
+)
+
+# Set agent
+camel_agent = ChatAgent(
+    system_message=sys_msg,
+    model=model,
+    tools=tools,
+)
+camel_agent.reset()
+
+# Define a user message
+usr_msg = BaseMessage.make_user_message(
+    role_name="CAMEL User",
+    content="Lists all pages in the Notion workspace",
+)
+
+# Get response information
+response = camel_agent.step(usr_msg)
+print(str(response.info['tool_calls'])[:1000])
+"""
+==========================================================================
+[FunctionCallingRecord(func_name='list_all_pages', args={}, result=[{'id': 
+'12684f56-4caa-8080-be91-d7fb1a5834e3', 'title': 'test page'}, 
+{'id': '47a4fb54-e34b-4b45-9928-aa2802982eb8', 'title': 'Aigentbot'}])]
+"""
+
+usr_msg = BaseMessage.make_user_message(
+    role_name="CAMEL User",
+    content="Lists all pages in the Notion workspace",
+)
+
+# Get response information
+response = camel_agent.step(usr_msg)
+print(str(response.info['tool_calls'])[:1000])
+"""
+==========================================================================
+[FunctionCallingRecord(func_name='list_all_pages', args={}, result=[{'id': 
+'12684f56-4caa-8080-be91-d7fb1a5834e3', 'title': 'test page'}, 
+{'id': '47a4fb54-e34b-4b45-9928-aa2802982eb8', 'title': 'Aigentbot'}])]
+"""
```

Comment:
redundant

Reply:
This example is not redundant. There are three examples in total. This one is asking the agent to return a list of pages in the notion workspace.